### ISSUE TITLE
config.core.php overwriten with absolute path on update

### ISSUE THREAD

**Issue by pepebe** [modxcms/revolution#12370]
This might be related to https://github.com/modxcms/revolution/issues/7366

I just updated a modx installation with a traditional modx package. During setup the all config.core.php files are rewritten to include the absolute path.
```
define('MODX_CORE_PATH', '/home/pepebe/subdomains/test1/core/');
````
As during installation is perfectly fine with 

```
define('MODX_CORE_PATH', dirname(dirname(__FILE__)) . '/core/');
```

I don't see the point, why a later update should change those values without asking for a permission.

This causes HUGE problems with cloning installations.

Cheers,

pepebe

**Comment by inreti** [comment]
If the original value would be left, you get problems with hardened installations, where the core is moved and/or renamed.

**Comment by christianseel** [comment]
Confirming. We also use the config files with relative paths across our projects and this is annoying after every MODX update.

**Comment by mindeffects** [comment]
OK, @wuuti made a good point here. Makes sense.

#modxbughunt

**Comment by christianseel** [comment]
The setup process needs to check if the path has been changed during update and if it's still reachable. If it didn't changes and is still correct, there is no reason why the config files should be updated.

**Comment by mindeffects** [comment]
That is correct for setup round 2+. Once in place, those files should not change.

Still, for migrating/moving a website, it would come handy if the setup would test the path and only "correct" the files if needed. If everything is in place, it shouldn't do anything.

#modxbughunt

**Comment by OptimusCrime** [comment]
It could perhaps be possible to attempt to evaluate the content of the file to see if the defined constant is valid or not?
